spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -454,6 +503,13 @@ private boolean tryRedisLock(long time) throws ExecutionException, InterruptedEx
 					LOGGER.debug("Acquired lock; " + this);
 				}
 				this.lockedAt = System.currentTimeMillis();
+				if (RedisLockRegistry.this.renewalScheduledExecutor != null) {
+					long delay = RedisLockRegistry.this.expireAfter / 3;
+					Runnable task = () -> {
+						renewLock(this.lockKey.substring((RedisLockRegistry.this.registryKey + ':').length()));
+					};
+					RedisLockRegistry.this.renewalScheduledExecutor.scheduleWithFixedDelay(task, delay, delay, TimeUnit.MILLISECONDS);
```

Comment:
You are missing extracting a `volatile ScheduledFuture<?>` for this call to `cancel(true)` when we `unlock()`.